src/haz3lcore/lang/term/Grammar.re

```diff
@@ -415,3 +415,385 @@ and map_type_hole_annotation:
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let ap = (~ann=?, d, e1, e2): exp_t(DefaultAnnotation.t) => {
+    term: Ap(d, e1, e2),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let typ_ap = (~ann=?, e, t): exp_t(DefaultAnnotation.t) => {
+    term: TypAp(e, t),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let deferred_ap = (~ann=?, e, l): exp_t(DefaultAnnotation.t) => {
+    term: DeferredAp(e, l),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let if_ = (~ann=?, e1, e2, e3): exp_t(DefaultAnnotation.t) => {
+    term: If(e1, e2, e3),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let seq = (~ann=?, e1, e2): exp_t(DefaultAnnotation.t) => {
+    term: Seq(e1, e2),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let test = (~ann=?, e): exp_t(DefaultAnnotation.t) => {
+    term: Test(e),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let filter = (~ann=?, k, e): exp_t(DefaultAnnotation.t) => {
+    term: Filter(k, e),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let closure = (~ann=?, env, e): exp_t(DefaultAnnotation.t) => {
+    term: Closure(env, e),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let parens = (~ann=?, e): exp_t(DefaultAnnotation.t) => {
+    term: Parens(e),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let cons = (~ann=?, e1, e2): exp_t(DefaultAnnotation.t) => {
+    term: Cons(e1, e2),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let list_concat = (~ann=?, e1, e2): exp_t(DefaultAnnotation.t) => {
+    term: ListConcat(e1, e2),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let un_op = (~ann=?, op, e): exp_t(DefaultAnnotation.t) => {
+    term: UnOp(op, e),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let bin_op = (~ann=?, op, e1, e2): exp_t(DefaultAnnotation.t) => {
+    term: BinOp(op, e1, e2),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let builtin_fun = (~ann=?, s): exp_t(DefaultAnnotation.t) => {
+    term: BuiltinFun(s),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let match = (~ann=?, e, l): exp_t(DefaultAnnotation.t) => {
+    term: Match(e, l),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+  let cast = (~ann=?, e, t1, t2): exp_t(DefaultAnnotation.t) => {
+    term: Cast(e, t1, t2),
+    annotation: Option.value(~default=DefaultAnnotation.default_value(), ann),
+  };
+
+  // pat
+  let pat_invalid = (~ann=?, s): pat_t(DefaultAnnotation.t) => {
```

Comment:
We can also nest modules inside here instead of prepending sort. This works with local open so you could do:

```reason
FTemp.(Exp.(list([int(1), int(2), cast(int(3), Typ.int(), Typ.unknown(Internal)))))
```